action.yaml

```diff
@@ -0,0 +1,358 @@
+---
+name: Setup R dependencies
+author: Insights Engineering
+description: GitHub Action to install R package dependencies
+
+inputs:
+  github-token:
+    description: GitHub token.
+    required: false
+    default: ""
+  repository-path:
+    description: Directory where the R package is located relative to the calling GitHub workflow workspace.
+    required: false
+    default: "."
+  lookup-refs:
+    description: |
+      List of package references to be used for the feature branch.
+      Multiple entries in new lines or separated by commas.
+    required: false
+    default: ""
+  extra-deps:
+    description: |
+      List of extra dependencies to check against feature branch existence.
+      Multiple entries in new lines or separated by commas. It allows to control indirect dependencies.
+      This requires an entry in the `lookup-refs` input.
+    required: false
+    default: ""
+  dependencies:
+    description: Passed to `r-lib/actions/setup-r-dependencies`.
+    required: false
+    default: '"all"'
+  extra-packages:
+    description: Passed to `r-lib/actions/setup-r-dependencies`.
+    required: false
+    default: ""
+  needs:
+    description: |
+      Passed to `r-lib/actions/setup-r-dependencies`.
+      The value will be amended by `DepsBranch` and `DepsDev` values.
+    required: false
+    default: ""
+  skip-desc-branch:
+    description: Skip the DESCRIPTION file modification for the feature branch.
+    required: false
+    default: false
+  skip-desc-dev:
+    description: Skip the DESCRIPTION file modification for the development branch.
+    required: false
+    default: false
+  skip-install:
+    description: Skip the installation of R package dependencies.
+    required: false
+    default: false
+
+branding:
+  icon: 'download'
+  color: 'orange'
+
+runs:
+  using: composite
+  steps:
+
+  - name: Get branch names
+    id: branch-names
+    uses: tj-actions/branch-names@v8
+
+  - name: Install required packages
+    uses: r-lib/actions/setup-r-dependencies@v2
+    with:
+      packages: |
+        any::sessioninfo
+      extra-packages: |
+        any::pkgdepends
+        any::cli
+        any::desc
+      install-pandoc: false
+
+  - name: Prepare DESCRIPTION file (feature branch)
+    if: >
+      inputs.skip-desc-branch == 'false' ||
+      steps.branch-names.outputs.is_default == 'true'
+    id: desc-branch
+    shell: Rscript {0}
+    run: |
+      cat("::group::Define utils\n")
+      base_packages <- c("R", "base", "utils", "stats", "graphics", "grDevices", "datasets", "methods", "tools", "parallel", "compiler")
+      desc_field_create <- function(d, key) {
+        d$set_list(key, character(0))
+      }
+      desc_field_append <- function(d, key, value) {
+        old_values <- d$get_field(key)
+        new_values <- paste0(old_values, value, collapse = ", ")
```

Comment:
I think we want to use `paste` with `sep` parameter instead of `paste0`

``` r
old_values <- "averissimo/pkg3@custom"
value <- "pkg4=averissimo/pkg4@custom"
paste0(old_values, value, collapse = ", ")
#> [1] "averissimo/pkg3@custompkg4=averissimo/pkg4@custom"
```

``` r
old_values <- ""
value <- "averissimo/pkg3@custom"
paste(old_values, value, sep = ", ")
#> [1] ", averissimo/pkg3@custom"
```

<sup>Created on 2024-06-27 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

```suggestion
        new_values <- if (identical(trimws(old_values), "")) {
          value
        } else {
          paste(old_values, value, sep = ", ")
        }
```

Reply:
abc871e711dfe578e954bfd910576b8381eaade2